pallets/pallet-unicks/src/mock.rs

```diff
@@ -0,0 +1,204 @@
+
+parameter_types! {
+	pub const SS58Prefix: u8 = 38;
+	pub const BlockHashCount: BlockNumber = 2400;
+}
+
+impl frame_system::Config for Test {
+	type BaseCallFilter = frame_support::traits::Everything;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type Origin = Origin;
+	type Call = Call;
+	type Index = Index;
+	type BlockNumber = BlockNumber;
+	type Hash = Hash;
+	type Hashing = BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = Event;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = SS58Prefix;
+	type OnSetCode = ();
+}
+
+parameter_types! {
+	pub const ExistentialDeposit: Balance = 10;
+	pub const MaxLocks: u32 = 50;
+	pub const MaxReserves: u32 = 50;
+}
+
+impl pallet_balances::Config for Test {
+	type Balance = Balance;
+	type DustRemoval = ();
+	type Event = Event;
+	type ExistentialDeposit = ExistentialDeposit;
+	type AccountStore = System;
+	type MaxLocks = MaxLocks;
+	type MaxReserves = MaxReserves;
+	type ReserveIdentifier = [u8; 8];
+	type WeightInfo = ();
+}
+
+pub(crate) type TestUnick = AsciiUnick<Test, MinUnickLength, MaxUnickLength>;
+pub(crate) type TestUnickOwner = SubjectId;
+pub(crate) type TestUnickPayer = AccountId;
+pub(crate) type TestRegularOrigin = mock_origin::EnsureDoubleOrigin<TestUnickPayer, TestUnickOwner>;
+pub(crate) type TestOriginSuccess = mock_origin::DoubleOrigin<TestUnickPayer, TestUnickOwner>;
+pub(crate) type TestBanOrigin = EnsureRoot<AccountId>;
+
+parameter_types! {
+	pub const MaxUnickLength: u32 = 32;
+	pub const MinUnickLength: u32 = 3;
+	// Easier to setup insufficient funds for deposit but still above existential deposit
+	pub const UnickDeposit: Balance = 2 * ExistentialDeposit::get();
+}
+
+impl pallet_unicks::Config for Test {
+	type BanOrigin = TestBanOrigin;
+	type Currency = Balances;
+	type Deposit = UnickDeposit;
+	type Event = Event;
+	type MaxUnickLength = MaxUnickLength;
+	type MinUnickLength = MinUnickLength;
+	type OriginSuccess = TestOriginSuccess;
+	type RegularOrigin = TestRegularOrigin;
+	type Unick = TestUnick;
+	type UnickOwner = TestUnickOwner;
+	type WeightInfo = ();
+}
+
+impl mock_origin::Config for Test {
+	type Origin = Origin;
+	type AccountId = AccountId;
+	type SubjectId = SubjectId;
+}
+
+pub(crate) const ACCOUNT_00: TestUnickPayer = AccountId::new([1u8; 32]);
+pub(crate) const ACCOUNT_01: TestUnickPayer = AccountId::new([2u8; 32]);
+pub(crate) const DID_00: TestUnickOwner = SubjectId(ACCOUNT_00);
+pub(crate) const DID_01: TestUnickOwner = SubjectId(ACCOUNT_01);
+
+pub(crate) fn unick_00() -> TestUnick {
+	AsciiUnick::try_from(b"unick_00".to_vec()).unwrap()
+}
+pub(crate) fn unick_01() -> TestUnick {
+	AsciiUnick::try_from(b"unick_01".to_vec()).unwrap()
+}
```

Comment:
can we have a single function for this and make `unick_00` and `unick_01` constants? Adding a third unick is than just either use a third constant or use a local value.

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/314/commits/9697acb36d39dbf3067a469f0c6abf82bbc6e32f?

Reply:
Thanks!